app.js

```diff
@@ -0,0 +1,928 @@
+			}, Error);
+		})
+
+		it('should throw error if num > length of list', () => {
+
+			// if i is not a number
+			chai.assert.throws(() => {
+				removeNItems(1, 8, [])
+			}, Error);
+		})
+	});
+
+	/* 7
+		@function smartRemoveItems
+		@param i {number}
+		@param list {array, []}
+		@returns list
+		@description
+			- if `i` is < 0, remove i number of items
+				from END of list
+			- if `i` > length of list, return list immediately
+			- if `i` > 0 remove i number of items
+				from START of list
+	*/
+
+	// implement function here
+	const smartRemoveItems = (i, list = []) => {
+		if(i < 0) {
+			// given value that does not exist so we absolute.
+			list.splice(i, Math.abs(i));
+			return list
+		} else if(i > list.length) {
+			return list
+		} else {
+			list.splice(0, i);
+			return list;
+		}
+}
+
+	// TEST
+	describe('7. smartRemoveItems', () => {
+		it('should return list if i > length of list', () => {
+			let list = [];
+			list = smartRemoveItems(1, list);
+
+			// [] is initial state of list
+			// we expect `list` to also be length 0
+			// ...or, empty essentially
+			chai.assert.equal(list.length, 0);
+		});
+
+		it('should remove i number from end of list if i < 0', () => {
+			let list = addToShoppingList({
+				'item': 'test',
+				'price': 1
+			});
+			list = addToShoppingList({
+				'item': 'test2',
+				'price': 2
+			}, list);
+			list = addToShoppingList({
+				'item': 'test3',
+				'price': 3
+			}, list);
+
+			list = smartRemoveItems(-1, list);
+
+			chai.assert.equal(list.length, 2)
+			console.log(list);
+			chai.assert.equal(list[0].item, 'test')
+			chai.assert.equal(list[0].price, 1)
+			chai.assert.equal(list[1].item, 'test2')
+			chai.assert.equal(list[1].price, 2)
+		});
+
+		it('should remove i number from START if list if i > 0', () => {
+			let list = addToShoppingList({
+				'item': 'test',
+				'price': 1
+			});
+			list = addToShoppingList({
+				'item': 'test2',
+				'price': 2
+			}, list);
+			list = addToShoppingList({
+				'item': 'test3',
+				'price': 3
+			}, list);
+
+			list = smartRemoveItems(1, list);
+
+			chai.assert.equal(list.length, 2)
+			chai.assert.equal(list[0].item, 'test2')
+			chai.assert.equal(list[0].price, 2)
+			chai.assert.equal(list[1].item, 'test3')
+			chai.assert.equal(list[1].price, 3)
+		});
+	});
+
+	/* 8
+		@function spliceItem
+		@para item {object}
+		@param i {number}
+		@param list {array, []}
+		@returns list
+		@description
+			- item must be an object that looks like this:
+			{
+				'item': 'eggs',
+				'price': 1.59
+			} (else throw error)
+			- insert item into the ith index of the list
+			- if i > length of list, just append
+			- if i < 0, just prepend
+	*/
+
+	// implement function here
+	const spliceItem = (item, i, list = []) => {
+		if(!item.hasOwnProperty('item') || !item.hasOwnProperty('price')){
+			throw new Error('This is not valid')
+		} else if(i > list.length) {
+			list.push(item);
+		} else if(i < 0) {
+			list.unshift(item);
+		}
```

Comment:
Don't forget you need a `return` statement here.

Consider this finished function that also takes into account [this instruction](https://github.com/FEWDMaterials/JSR_Shopping_List_Functions/pull/18/files#diff-0364f57fbff2fabbe941ed20c328ef1aR453)

```
const spliceItem = (item, i, list = []) => {
		if(!item.hasOwnProperty('item') || !item.hasOwnProperty('price')){
			throw new Error('This is not valid')
		} else if(i > list.length) {
			list.push(item);
		} else if(i < 0) {
			list.unshift(item);
		} else {
			list.splice(i,0,item);
		}

		return list;
	}
```